Sources/TuistKit/Services/GraphService.swift

```diff
@@ -41,19 +47,29 @@ final class GraphService {
     {
         let graph = try manifestGraphLoader.loadGraph(at: path)
 
-        let graphVizGraph = graphVizMapper.map(
-            graph: graph,
-            skipTestTargets: skipTestTargets,
-            skipExternalDependencies: skipExternalDependencies,
-            targetsToFilter: targetsToFilter
-        )
-
         let filePath = outputPath.appending(component: "graph.\(format.rawValue)")
         if FileHandler.shared.exists(filePath) {
             logger.notice("Deleting existing graph at \(filePath.pathString)")
             try FileHandler.shared.delete(filePath)
         }
-        try export(graph: graphVizGraph, at: filePath, withFormat: format, layoutAlgorithm: layoutAlgorithm)
+        
+        switch format {
+        case .dot, .png:
+            let graphVizGraph = graphVizMapper.map(
+                graph: graph,
+                skipTestTargets: skipTestTargets,
+                skipExternalDependencies: skipExternalDependencies,
+                targetsToFilter: targetsToFilter
+            )
+            
+            try export(graph: graphVizGraph, at: filePath, withFormat: format, layoutAlgorithm: layoutAlgorithm)
+        case .json:
+            let config = try configLoader.loadConfig(path: path)
+            
+            let outputGraph = GraphOutputJSON.from(graph, withConfig: config)
+            try outputGraph.export(to: filePath)
```

Comment:
I think we should make the `path` optional (don't default to `FileHandler.shared.currentPath` in `GraphCommand`) and output JSON to stdout by default. This will make it easier to reuse the output in a different process.

Reply:
That deviates from the other two formats though. Besides, stdout can have a number of outputs in the future. It seems less future-proof and more error-prone to rely on stdout.

Reply:
> It seems less future-proof and more error-prone to rely on stdout.

I don't understand this part. The way I see it is that `tuist graph --format json` will be mostly used by other processes using the output and it's therefore better as a default instead of a file.

Reply:
If somehow in the future a dependency of the command, say one of the mappers or loaders print to stdout, we would have broken the output. SPM actually had this exact same issue. I'm trying to avoid that mistake here.

Reply:
ah, ok, I understand now, you convinced me then 🙂

Reply:
Thank you for the suggestion!